controllers/articles-controller.js

```diff
@@ -0,0 +1,27 @@
+const {selectArticleById} = require('../models/articles-model')
+
+function getArticleById(req, res, next){
+
+    const articleId = req.params.article_id
+
+    selectArticleById(articleId)
+    .then((data) => {
+       
+        
+        if (data.rows.length !== 0) {
+            
+            const article = data.rows[0]
+            res.status(200).send({article})
+
+        } else {
+            res.status(404).send({message: 'Article does not exist'})
```

Comment:
Instead of sending a response here in case of an error, please deal with the error in the model, rejecting the promise with the appropriate object, then handle this and the response in your custom error handling middleware.